### PR TITLE
Hash username to avoid invalid file paths

### PR DESCRIPTION
**Pull request by ilikebits**
This resolves follow-up issues on #22 where some usernames create invalid file paths for the cache file.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,12 +5,16 @@
...
 const path = require('path');
+const crypto = require('crypto');
 const execFile = require('child_process').execFile;
 const env = process.env;
 const user = env.LOGNAME || env.USER || env.LNAME || env.USERNAME;
-const configfile = '.v8flags.'+process.versions.v8+'.'+user+'.json';
 const exclusions = ['--help'];
 
+const hash = crypto.createHash('md5');
+hash.update(user);
```
Comment (phated) [inline index.js]: I was pretty sure this was how md5 hashes were done in node, but I just tried it and it didn't work.  Can you please add a test for this also?  Thanks

Comment (phated) [inline index.js]: This looks to be due to encoding.

Comment (phated) [inline index.js]: Could you string together the calls, e.g. `crypto.createHash('md5').update(user, encoding).digest(encoding)`?

Comment (ilikebits) [inline index.js]: @phated, what version are you using? The default encoding works fine for me on Node 5.10.1.

Fixed the call.

Comment (ilikebits) [inline index.js]: Also, the default tests still work because they depend on the `configfile` exported by this file.

**Comment by ilikebits** [comment]
@phated does this look good to merge?

**Comment by phated** [comment]
@ilikebits can you add a test where `env.USER` is set to something that would produce an invalid path on the filesystem?  That's the one thing that is still outstanding.

**Comment by preppypiet** [comment]
Could you please complete this PR? Ran into this issue again today and this would fix it perfectly. Instead I had to run a custom script with code I grabbed from this lib to manually make the directory in case username had a / in it.

**Comment by phated** [comment]
@tkellen are we good to merge this?

**Comment by tkellen** [comment]
I'm gunshy given how many issues we've had with this project, but yes, I think we are.

**Comment by phated** [comment]
@tkellen me too, but I figured I'd be around my computer today and could field all the backlash 😛

**Comment by phated** [comment]
Published as 2.1.0 (seems like a feature and to give more wiggle room in semver).

**Comment by phated** [comment]
@tkellen gunshy not in vain, but I have it fixed with only a really weird, rare edge case that might encounter issues (multiple accounts without USER env that end up with the same json config but not matching).

**Comment by tkellen** [comment]
Ha! Thank you for taking the brunt of this @phated

**Pull request merged**